src/function/scalar/list/list_sort.cpp

```diff
@@ -106,7 +106,12 @@ static void ListSortFunction(DataChunk &args, ExpressionState &state, Vector &re
 	result.SetVectorType(VectorType::FLAT_VECTOR);
 	auto &result_validity = FlatVector::Validity(result);
 
-	args.Flatten();
+	for (auto &v : args.data) {
+		if (v.GetVectorType() == VectorType::DICTIONARY_VECTOR) {
```

Comment:
Maybe this should check for `!=FLAT_VECTOR && !=CONSTANT_VECTOR` instead as you did in `src/function/scalar/list/array_slice.cpp`? There could be other vector types in the future that need to be flattened here? (compressed vectors for example)

Reply:
Good point!